datafusion/core/tests/sql/functions.rs

```diff
@@ -67,8 +67,8 @@ async fn csv_query_cast_literal() -> Result<()> {
         "+--------------------+----------+",
         "| c12                | Int64(1) |",
         "+--------------------+----------+",
-        "| 0.9294097332465232 | 1        |",
-        "| 0.3114712539863804 | 1        |",
+        "| 0.9294097332465232 | 1.0      |",
```

Comment:
I am confused about this changes？
cc @alamb @tustvold 
As you said in https://github.com/apache/arrow-datafusion/pull/5241/files#r1106342898.

Reply:
What are you confused about? Despite its name, this column is a floating point type and so is formatted with a decimal point?

Reply:
> Despite its name

I certainly found that confusing 😆

Reply:
> What are you confused about? Despite its name, this column is a floating point type and so is formatted with a decimal point?

I missed the data type from the name of this column